tests/integration_tests/functional/test_error_code.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests scenarios for Firecracker signal handling."""
+
+import os
+import time
+import platform
+import pytest
+
+
+@pytest.mark.skipif(
+    platform.machine() == "x86_64",
+    reason="We only test with a kernel for aarch64"
+)
+def test_enosys_error_code(test_microvm_with_ssh):
+    """Test that ENOSYS error_code doesn't kill the microVM."""
+    test_microvm = test_microvm_with_ssh
+    test_microvm.spawn()
+
+    # We don't need to monitor the memory for this test.
+    test_microvm.memory_monitor = None
+
+    test_microvm.kernel_file = \
+        test_microvm.kernel_file[:-11] + "enosys_vmlinux.bin"
+    test_microvm.basic_config(vcpu_count=1)
+
+    test_microvm.start()
+
+    firecracker_pid = test_microvm.jailer_clone_pid
+
+    while True:
```

Comment:
I've seen this logic in multiple tests. It would be nice to have a helper method for it. Not blocking the PR for this, though.

Reply:
You can use `wait_process_termination` from ``framework.utils`